controllers/databaseclusterbackup_controller.go

```diff
@@ -153,25 +160,206 @@ func (r *DatabaseClusterBackupReconciler) SetupWithManager(mgr ctrl.Manager) err
 	if err == nil {
 		if err := r.addPXCToScheme(r.Scheme); err == nil {
 			controller.Owns(&pxcv1.PerconaXtraDBClusterBackup{})
+			controller.Watches(
+				&pxcv1.PerconaXtraDBClusterBackup{},
+				handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+					return r.tryCreateDBBackups(ctx, obj, r.tryCreatePXC)
+				}))
 		}
 	}
 
 	err = r.Get(ctx, types.NamespacedName{Name: psmdbBackupCRDName}, unstructuredResource)
 	if err == nil {
 		if err := r.addPSMDBToScheme(r.Scheme); err == nil {
 			controller.Owns(&psmdbv1.PerconaServerMongoDBBackup{})
+			controller.Watches(
+				&psmdbv1.PerconaServerMongoDBBackup{},
+				handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+					return r.tryCreateDBBackups(ctx, obj, r.tryCreatePSMDB)
+				}))
 		}
 	}
 
 	err = r.Get(ctx, types.NamespacedName{Name: pgBackupCRDName}, unstructuredResource)
 	if err == nil {
 		if err := r.addPGToScheme(r.Scheme); err == nil {
 			controller.Owns(&pgv2.PerconaPGBackup{})
+			controller.Watches(
+				&pgv2.PerconaPGBackup{},
+				handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+					return r.tryCreateDBBackups(ctx, obj, r.tryCreatePG)
+				}))
 		}
 	}
 	return controller.Complete(r)
 }
 
+func (r *DatabaseClusterBackupReconciler) tryCreateDBBackups(
+	ctx context.Context,
+	obj client.Object,
+	createBackupFunc func(ctx context.Context, obj client.Object) error,
+) []reconcile.Request {
+	logger := log.FromContext(ctx)
+	if len(obj.GetOwnerReferences()) == 0 {
+		err := createBackupFunc(ctx, obj)
+		if err != nil {
+			logger.Error(err, fmt.Sprintf("Failed to create DatabaseClusterBackup %s", obj.GetName()))
+		}
+	}
+
+	return []reconcile.Request{{
+		NamespacedName: types.NamespacedName{
+			Name:      obj.GetName(),
+			Namespace: obj.GetNamespace(),
+		},
+	}}
```

Comment:
This may not be needed, let's check.
Since the `createBackupFunc` creates a `DatabaseClusterBackup` resource, the reconcile loop shall be triggered automatically.

Reply:
I agree that this isn't needed for the reason you mentioned.

Moreover, I also think this is functionally wrong. We are queuing a request for a `pxcv1.PerconaXtraDBClusterBackup`, `psmdbv1.PerconaServerMongoDBBackup` or `pgv2.PerconaPGBackup` resource and not the `DatabaseClusterBackup` which is definitely not what we want.

Reply:
We spoke on the phone with Oksana and we need to trigger reconciliation here.
My confusion was from the fact I assumed we only need to create the `DatabaseClusterBackup` resource, but we also need to update its status. For this reason it's necessary.

Reply:
You guys are correct.

Before, we didn't need the watch because we set the controller reference on the `pxcv1.PerconaXtraDBClusterBackup`, `psmdbv1.PerconaServerMongoDBBackup` and `pgv2.PerconaPGBackup` when we create them and that automatically watches for changes in these resources and triggers a reconciliation of the `DatabaseClusterBackup`.
Now that we explicitly set a watcher, it's overriding the one automatically set when set the controller reference. So we do in fact need to return the list of reconcile requests.

However, I believe that we should return the reference to the `DatabaseClusterBackup` which is what we want to be reconciled and not the `pxcv1.PerconaXtraDBClusterBackup`, `psmdbv1.PerconaServerMongoDBBackup` or `pgv2.PerconaPGBackup` . Right?

Reply:
@recharte what do you mean? the `reconcile.Request` contains only the name and the namespace, there is no way to specify the kind of the resource. The controller is created for the `DatabaseClusterBackup` recources and this is the kind that is going to be reconciled. The controller watches the upstream recource, but the reconciling happening with the `DatabaseClusterBackup` only.

Reply:
Yes that's right, I got confused when I saw you were using the name of the upstream backup but you're completely correct, the controller controls the `DatabaseClusterBackup` so that's the resource kind he'll be on the lookout for.
Please disregard my comment, I messed up.